root/static/scripts/common/components/Relationship.js

```diff
@@ -0,0 +1,178 @@
+              : exp.l('in {date}', {date: formattedBeginDate})
+          ) : (
+            exp.l(
+              'from {begin_date} until {end_date}',
+              {
+                begin_date: formattedBeginDate,
+                end_date: formattedEndDate,
+              },
+            )
+          )
+        ) : (
+          relationship.ended
+            ? exp.l('from {date} to ????', {date: formattedBeginDate})
+            : exp.l('from {date} to present', {date: formattedBeginDate})
+        )}
+      </>
+    ) : (
+      hasEndDate ? (
+        <>
+          {' '}
+          {exp.l('until {date}', {date: formattedEndDate})}
+        </>
+      ) : (
+        relationship.ended ? (
+          <>
+            {' '}
+            {bracketedText(l('ended'))}
+          </>
+        ) : null
+      )
+    )
+  );
+};
+
+const HistoricRelationshipContent = ({
+  relationship,
+}: {relationship: RelationshipT}) => {
+  const linkType = linkedEntities.link_type[relationship.linkTypeID];
+  const extraAttributes = getExtraAttributes(
+    relationship,
+    'link_phrase',
+    false, /* forGrouping */
+  );
+  return (
+    <>
+      {linkType.link_phrase /* hardcoded untranslatable historical hack */}
+      {' '}
+      <DescriptiveLink
+        // Used for historic edits which hardcode entity1 as target
+        content={relationship.entity1_credit}
+        entity={relationship.target}
+      />
+      <RelationshipDates relationship={relationship} />
+      {extraAttributes ? (
+        <>
+          {' '}
+          {bracketedText(commaOnlyListText(extraAttributes.map(
+            attribute => attribute.typeName,
+          )))}
+        </>
+      ) : null}
+    </>
+  );
+};
+
+const RelationshipContent = ({
+  relationship,
+}: {relationship: RelationshipT}) => {
+  const direction = relationship.direction;
+  const linkType = linkedEntities.link_type[relationship.linkTypeID];
+  let entity0;
+  let entity1;
+  if (direction === 'backward') {
+    entity0 = relationship.target;
+    entity1 = linkedEntities[linkType.type1][relationship.entity1_id];
+  } else {
+    entity1 = relationship.target;
+    entity0 = linkedEntities[linkType.type0][relationship.entity0_id];
+  }
+  const longPhrase = interpolate(
+    relationship,
+    'long_link_phrase',
+    false /* forGrouping */,
+    <DescriptiveLink
+      content={relationship.entity0_credit}
+      entity={entity0}
+    />,
+    <DescriptiveLink
+      content={relationship.entity1_credit}
+      entity={entity1}
+    />,
+  );
+  const extraAttributes = getExtraAttributes(
+    relationship,
+    'long_link_phrase',
+    false, /* forGrouping */
+  );
+  return (
+    <>
+      {longPhrase}
+      {' '}
+      <RelationshipDates relationship={relationship} />
+      {extraAttributes ? (
+        <>
+          {' '}
+          {bracketedText(commaOnlyListText(extraAttributes.map(
+            attribute => attribute.typeName,
+          )))}
```

Comment:
Also looks like this should use root/static/scripts/common/utility/displayLinkAttribute.js